src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/jvm/ThreadList.java

```diff
@@ -104,37 +111,55 @@ public String toString() {
     }
   }
 
+  /**
+   * This is called from OSMetricsCollector#collectMetrics. So this is not called
+   * in the critical path of ES request handling. Even for the collector thread,
+   * we do a timed wait to acquire this lock and move on if we could not get it.
+   * @return A hashmap of threadId to threadState.
+   */
   public static Map<Long, ThreadState> getNativeTidMap() {
-    synchronized (ThreadList.class) {
-      if (System.currentTimeMillis() > lastRunTime + minRunInterval) {
-        runThreadDump(pid, new String[0]);
+    final int waitTimeOutMillis = 100;
+    try {
+      if (vmAttachLock.tryLock(waitTimeOutMillis, TimeUnit.MILLISECONDS)) {
+        try {
+          if (System.currentTimeMillis() > lastRunTime + minRunInterval) {
+            runThreadDump(pid, new String[0]);
```

Comment:
Lets add an assertion like it shouldn't be any of ES threads

Reply:
I added a metric to track such occurrences.